examples/normalizing_flows/dlgm_nf.py

```diff
@@ -29,14 +30,17 @@ def vae(observed, n, x_dim, z_dim, n_particles):
     return model
 
 
-def q_net(x, z_dim, n_particles):
+def q_net(x, z_dim, n_particles, n_flows):
+    def forward(samples):
+        return zs.repeated_flow(zs.planar_normalizing_flow, samples, n_iters=n_flows)
+
     with zs.BayesianNet() as variational:
         lz_x = tf.layers.dense(tf.to_float(x), 500, activation=tf.nn.relu)
         lz_x = tf.layers.dense(lz_x, 500, activation=tf.nn.relu)
         z_mean = tf.layers.dense(lz_x, z_dim)
         z_logstd = tf.layers.dense(lz_x, z_dim)
-        z = zs.Normal('z', z_mean, logstd=z_logstd, group_ndims=1,
-                      n_samples=n_particles)
+        z = zs.NormalFlow('z', forward, mean=z_mean, logstd=z_logstd, group_ndims=1,
```

Comment:
I feel `NormalFlow` is a bit too restricted. And it also makes the internal Normal node invisible. I'd prefer sth. like
```python
z_pre = zs.Normal('z_pre', ...)
z = zs.Flow('z', flow=zs.planar_normalizing_flow, n_iters=n_flows)
```

Reply:
However, this would require some extra internal machinery or require that the input to the flow is an instance of `StochastcTensor` in order to still be able to use its sampling and log-probability methods